linera-core/src/chain_worker/state/mod.rs

```diff
@@ -305,12 +305,12 @@ where
     }
 
     /// Returns an error if the block requires a blob we don't have.
-    /// Looks for the blob in: chain manager's pending blobs and storage.
+    /// Looks for the blob in: chain manager's proposed blobs and storage.
     async fn check_no_missing_blobs(
         &self,
         required_blob_ids: &HashSet<BlobId>,
     ) -> Result<(), WorkerError> {
```

Comment:
Not in this PR but instead of returning the `missing_blob_ids` as the error, we could instead return `Vec<BlobId>` from this method and it'd contain all the missing blobs.

Reply:
And then error out on the caller? I'm not sure I get it 🤔 We have retry logic that relies on the missing blobs being returned as this error, so we can't remove it AFAIU